testcrate/tests/count_leading_zeros.rs

```diff
@@ -1,23 +1,26 @@
-extern crate compiler_builtins;
+use rand::random;
 
 use compiler_builtins::int::__clzsi2;
 
 #[test]
 fn __clzsi2_test() {
-    let mut i: usize = core::usize::MAX;
-    // Check all values above 0
-    while i > 0 {
-        assert_eq!(__clzsi2(i) as u32, i.leading_zeros());
-        i >>= 1;
-    }
-    // check 0 also
-    i = 0;
-    assert_eq!(__clzsi2(i) as u32, i.leading_zeros());
-    // double check for bit patterns that aren't just solid 1s
-    i = 1;
-    for _ in 0..63 {
-        assert_eq!(__clzsi2(i) as u32, i.leading_zeros());
-        i <<= 2;
-        i += 1;
+    // binary fuzzer
+    let mut x = 0usize;
+    let mut ones: usize;
+    // creates a mask for indexing the bits of the type
+    let bit_indexing_mask = usize::MAX.count_ones() - 1;
+    for _ in 0..1000 {
+        for _ in 0..4 {
+            let r0: u32 = bit_indexing_mask & random::<u32>();
+            ones = !0 >> r0;
+            let r1: u32 = bit_indexing_mask & random::<u32>();
+            let mask = ones.rotate_left(r1);
+            match (random(), random()) {
```

Comment:
Tests should be deterministic. Otherwise they could fail one time and succed another time.

Reply:
Could I use a seedable RNG like rand_xoshiro::Xoroshiro128StarStar?

Reply:
I am also using this fuzzing technique in PR #332.

Reply:
I guess you could.

Reply:
Randomized testing is fine in my opinion but this should also print out the input on failure so if it happens in CI it can be reproduced.

Reply:
I wonder if using something like https://docs.rs/quickcheck/0.9.2/quickcheck/ something we can consider for tests?